src/mono/browser/runtime/invoke-cs.ts

```diff
@@ -6,126 +6,96 @@ import WasmEnableThreads from "consts:wasmEnableThreads";
-        const method = cwraps.mono_wasm_assembly_find_method(klass, wrapper_name, -1);
-        if (!method)
-            throw new Error(`Could not find method: ${wrapper_name} in ${klass} [${assembly}]`);
-
-        const arg_marshalers: (BoundMarshalerToCs)[] = new Array(args_count);
-        for (let index = 0; index < args_count; index++) {
-            const sig = get_sig(signature, index + 2);
-            const marshaler_type = get_signature_type(sig);
-            const arg_marshaler = bind_arg_marshal_to_cs(sig, marshaler_type, index + 2);
-            mono_assert(arg_marshaler, "ERR43: argument marshaler must be resolved");
-            arg_marshalers[index] = arg_marshaler;
-        }
 
-        const res_sig = get_sig(signature, 1);
-        let res_marshaler_type = get_signature_type(res_sig);
-        const is_async = res_marshaler_type == MarshalerType.Task;
-        if (is_async) {
-            res_marshaler_type = MarshalerType.TaskPreCreated;
-        }
-        const res_converter = bind_arg_marshal_to_js(res_sig, res_marshaler_type, 1);
-
-        const closure: BindingClosure = {
-            method,
-            fqn: js_fqn,
-            args_count,
-            arg_marshalers,
-            res_converter,
-            is_async,
-            isDisposed: false,
-        };
-        let bound_fn: Function;
-        // void
-        if (args_count == 0 && !res_converter) {
-            bound_fn = bind_fn_0V(closure);
-        }
-        else if (args_count == 1 && !res_converter) {
-            bound_fn = bind_fn_1V(closure);
-        }
-        else if (is_async && args_count == 1 && res_converter) {
-            bound_fn = bind_fn_1RA(closure);
-        }
-        else if (is_async && args_count == 2 && res_converter) {
-            bound_fn = bind_fn_2RA(closure);
-        }
-        else if (args_count == 1 && res_converter) {
-            bound_fn = bind_fn_1R(closure);
-        }
-        else if (args_count == 2 && res_converter) {
-            bound_fn = bind_fn_2R(closure);
-        }
-        else {
-            bound_fn = bind_fn(closure);
-        }
+    const version = get_signature_version(signature);
+    mono_assert(version === 2, () => `Signature version ${version} mismatch.`);
 
-        // this is just to make debugging easier. 
-        // It's not CSP compliant and possibly not performant, that's why it's only enabled in debug builds
-        // in Release configuration, it would be a trimmed by rollup
-        if (BuildConfiguration === "Debug" && !runtimeHelpers.cspPolicy) {
-            try {
-                bound_fn = new Function("fn", "return (function JSExport_" + methodname + "(){ return fn.apply(this, arguments)});")(bound_fn);
-            }
-            catch (ex) {
-                runtimeHelpers.cspPolicy = true;
-            }
-        }
+    const { assembly, namespace, classname, methodname } = parseFQN(fullyQualifiedName);
```

Comment:
Can we pass the parsed values to C#, so that don't need to replicate the logic there?